src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -21,33 +24,26 @@ internal sealed class NoValue
     public static NoValue Instance = new();
 }
 
-internal class QueueItem<TRequest, TResponse, TRequestContext> : IQueueItem<TRequestContext>
+internal class QueueItem<TRequestContext> : IQueueItem<TRequestContext>
 {
-    private readonly TRequest _request;
-
     private readonly ILspLogger _logger;
     private readonly AbstractRequestScope? _requestTelemetryScope;
 
     /// <summary>
     /// A task completion source representing the result of this queue item's work.
     /// This is the task that the client is waiting on.
     /// </summary>
-    private readonly TaskCompletionSource<TResponse> _completionSource = new();
+    private readonly TaskCompletionSource<object?> _completionSource = new();
```

Comment:
i missed why this got more weakly typed.

Reply:
The QueueItem is created when the request is added to the queue.  Previously we knew the concrete types (and so it was QueueItem<TRequest, TResponse>, but now we don't.

Reply:
i get taht we don't know the language.  i don't quite get why we don't know the TREquest/TResponse types.  Is that lang dependent somehow?